### PR TITLE
Fixed Returning 401 on any unauthorised routes #47

### PR DESCRIPTION
**Pull request by kousikmitra**
Implemented a method called `_is_route_exist` in `_routing.py` which checks whether a particular route exist in the current app to handle the requested URL.
If a router exists it returns `True` else returns `False`
Depending on the result the further processing is done as before.

### CODE REVIEW

File: flask_jwt_router/_routing.py [hunk 1]
```diff
@@ -116,26 +117,40 @@ def _allow_public_routes(self, white_routes):
                 return False
         return True
 
+    def _is_route_exist(self, url: str, method: str) -> bool:
```
Comment (joegasewicz) [inline flask_jwt_router/_routing.py]: can we name this method: `_does_route_exist` please.

File: flask_jwt_router/_routing.py [hunk 2]
```diff
@@ -116,26 +117,40 @@ def _allow_public_routes(self, white_routes):
...
-            ignored_routes = self.extensions.ignored_routes
-            if len(ignored_routes):
-                is_ignored = not self._allow_public_routes(ignored_routes)
-            if not is_ignored:
-                white_routes = self._prefix_api_name(self.extensions.whitelist_routes)
-                not_whitelist = self._allow_public_routes(white_routes)
-
-                if not_whitelist:
-                    self._handle_token()
+            if self._is_route_exist(path, method):
```
Comment (joegasewicz) [inline flask_jwt_router/_routing.py]: rename to `_does_route_exist`

File: .gitignore [hunk 3]
```diff
@@ -109,6 +109,9 @@ venv_one/
 venv_one.bak/
 .idea/
 
+# VS Code project settings
+.vscode
```
Comment (joegasewicz) [inline .gitignore]: nice haha

File: flask_jwt_router/_routing.py (hunk 1 above)
Comment (kousikmitra) [reply flask_jwt_router/_routing.py]: Renamed

File: flask_jwt_router/_routing.py (hunk 2 above)
Comment (kousikmitra) [reply flask_jwt_router/_routing.py]: Renamed

**Comment by joegasewicz** [comment]
Sorry @kousikmitra, can you run `pipenv uninstall flake8` because we have an issue already for linting, thanks

File: .gitignore (hunk 3 above)
Comment (kousikmitra) [reply .gitignore]: The only product of MS I love. :p

**Comment by joegasewicz** [comment]
> > @app.route("/test", methods=["GET"])
> > def test_one():
> > return "/test"
> 
> Yep. It fixed the issue. What was the problem?

We need to register a route to the Flask app instance as those tests were just testing the before_middleware public method

**Comment by joegasewicz** [comment]
Great work @kousikmitra

**Comment by joegasewicz** [comment]
Let me know if you have any problems with the pipfile.lock but otherwise looks good

**Comment by joegasewicz** [comment]
Sorry @kousikmitra can we remove the `url_for` thanks

**Comment by joegasewicz** [comment]
its ok, i fixed the pipfile.lock, thanks

**Comment by joegasewicz** [comment]
@kousikmitra we are all good now, thank you again

**Comment by joegasewicz** [comment]
@kousikmitra if you want to have 100% checks have a look at this:

Screenshot 2019-12-02 at 16 31 36

**Pull request merged**